File: Sep24/btq59.py
# Code by Nehan Mohammad
# Gate 2025, Biotechnology, question 59.

# Given Parameters
power_limit = 10

# Calculate sequence iteratively
a = 0
n = 0
threshold = 1 / (2 ** power_limit)

while abs(1 - a) >= threshold:
    a = 0.5 * (1 + a)
    n += 1

print(f" Power limit exponent: {power_limit},\n Required value <= 1/2^10")
print(f"Least value of n: {n}")

# print(f"Absolute difference |1 - a_n|: {abs(1 - a):.8f}")

File: Sep24/btq59.c
// Code by Nehan Mohammad
// Gate 2025, Biotechnology, question 59.

#include <stdio.h>
#include <math.h>

int main() {
    // Given Parameters
	
    int power_limit = 10;
    
    // Calculate sequence iteratively
    double a = 0.0;
    int n = 0;
    double threshold = 1.0 / pow(2.0, power_limit);

    while (fabs(1.0 - a) >= threshold) {
        a = 0.5 * (1.0 + a);
        n++;
    }

    printf("Power limit exponent: %d,\n Threshold value: %e, \n Final sequence value of a_n: %.6f\n", power_limit, threshold, a);
    printf("Least value of n: %d\n", n);
    //printf("Absolute difference |1 - a_n|: %.8f\n", fabs(1.0 - a));

    return 0;
}

